webapp/src/ts/services/enketo.service.ts

```diff
@@ -570,6 +571,7 @@ export class EnketoService {
       .map((idx, element) => {
         const docToStore:any = this.enketoTranslationService.reportRecordToJs(getOuterHTML(element));
         docToStore._id = getId(Xpath.getElementXPath(element));
+        docToStore.xmlVersion = xmlVersion;
```

Comment:
I know this is a draft, but I thought I'd throw a comment in...

This is setting the `xmlVersion` in the child docs being created by the report but not in the report itself. This is potentially useful but we don't store anything about how the child doc was created, not even the form ID, so it's potentially not useful on its own. The primary place this should be set is in the `create` function where the report itself is created, right beside where we set the `form` property.

For CouchDB properties we [use snake case](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/#use-snake_case-for-couchdb-document-property-names) for naming (like `reported_date` below). This applies to both the form and report docs. There are some couterexamples you can find but we're standardising at least on _new_ properties.

Finally I'm not sure `xmlVersion` will make much sense to someone looking at the report who doesn't know the context. I think `form_version` would be better and be obvious when stored beside `form` (the form's ID).

Reply:
Thanks. So I understand, the change I made in https://github.com/medic/cht-conf/pull/431 will cause the `get(internalId)` function [here](https://github.com/medic/cht-core/blob/5ae422eb51b2e9bdf1fac49def562387a193348a/webapp/src/ts/services/xml-forms.service.ts#L237) to return a doc Promise that contains the new field `xmlVersion` that I add in https://github.com/medic/cht-conf/pull/431

Is that correct?

Reply:
Yes it should do! You'll obviously need to get the CHT running in a developer environment to test this end to end.